uqcsbot/cowsay.py

```diff
@@ -0,0 +1,141 @@
+
+from typing import Optional, Literal
+
+import discord
+from discord import app_commands
+from discord.ext import commands
+
+from uqcsbot.bot import UQCSBot
+
+# Type Alias for Cow Eyes 
+CowsayMoodType = Literal['Normal', 'Borg', 'Dead', 'Greed', 'Paranoid', 'Stoned', 'Tired', 'Wired', 'Youthful']
+
+class Cowsay(commands.Cog):
+
+    def __init__(self, bot: UQCSBot):
+        self.bot = bot
+        self._max_length = 40
+        self._cow_eyes = dict(
+            Normal = 'oo',
+            Borg = '==',
+            Dead = 'xx',
+            Greed = '$$',
+            Paranoid = '@@',
+            Stoned = '**',
+            Tired = '--',
+            Wired = 'OO',
+            Youthful = '..'
+        )
+
+    @app_commands.command(name="cowsay")
+    @app_commands.describe(
+        message="The message that the cow will say",
+        mood="The mood of the cow (optional)",
+        tongue="Whether the cow should show its tongue (optional, default to False)",
+        tux="Display the Linux Tux instead of the cow. Tux doesn't show tongue. (optional, default to False)",
+    )
+    async def cowsay_command(self, interaction: discord.Interaction, 
+            message: str, 
+            mood: Optional[CowsayMoodType] = 'Normal', 
+            tongue: Optional[bool] = False,
+            tux: Optional[bool] = False) -> None:
+        """
+        Returns a cow or tux saying the given message.
+        """
+
+        # Check message length, if invalid send moo!
+        if len(message) == 0 or len(message) > 2000:
```

Comment:
Cap this at 1000 characters, as its the final message it sends back that needs to be less than 2000 characters.
```suggestion
        if len(message) == 0 or len(message) > 1000:
```